### PR TITLE
chore(qa): Preparing 1st TheAnvil PROD release

### PR DESCRIPTION
**Pull request by themarcelor**
Preparing PR for TheAnvil PROD.

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
# gen3.theanvil.io/manifest.json
## :warning: Services on branch
- arranger-adminapi
- arranger-dashboard
- wts
## Deployment changes
- fence
  - Fence does NOT handle `c999` the same as before. If using dbgap sync w/ consent code parsing and indexd records have acls with just `phsid` to represent objects someone w/ `c999` should be able to see, you will need to ensure that the records also have the phsid.consent_code as well or users will lose access (uc-cdis/fence#702)
  - Fence has a new configuration to support more dbGaP syncing options: if you require "Common Exchange Area" data access, review the new configuration and supply necessary values (uc-cdis/fence#702)
  - Fence config ENABLED_IDENTITY_PROVIDERS section deprecated and replaced by LOGIN_OPTIONS section to allow enabling several login options for the same provider (backwards compatible) (uc-cdis/fence#703)
  - Requires manual work in Arborist DB to fill in the empty `authz_provider` fields (uc-cdis/fence#696)
  - new cloud automation job for database migration https://github.com/uc-cdis/cloud-automation/pull/977 (uc-cdis/fence#694)
- guppy
  - Commons admins can specify a selectedValues filter using the `adminPreAppliedFilters` field in a portal config file using Guppy syntax. The filter is applied to to the records displayed before any user-applied filters, and can be specified independently in the dataExplorerConfig and fileExplorerConfig blocks. Example: (uc-cdis/guppy#56)
  - ``` (uc-cdis/guppy#56)
  - "adminAppliedPreFilters": { (uc-cdis/guppy#56)
  - "project_id": { (uc-cdis/guppy#56)
  - "selectedValues": ["jnkns-jenkins"] (uc-cdis/guppy#56)
  - } (uc-cdis/guppy#56)
  - } (uc-cdis/guppy#56)
  - ``` (uc-cdis/guppy#56)
- hatchery
  - New `pull_policy` option is available for `hatchery.containers` entries, eg: (uc-cdis/hatchery#1)
  - ``` (uc-cdis/hatchery#1)
  - { (uc-cdis/hatchery#1)
  - "target-port": 8787, (uc-cdis/hatchery#1)
  - "cpu-limit": "1.0", (uc-cdis/hatchery#1)
  - "memory-limit": "512Mi", (uc-cdis/hatchery#1)
  - "pull_policy": "Always", (uc-cdis/hatchery#1)
  - "name": "R Studio", (uc-cdis/hatchery#1)
  - "image": "heliumdatastage/rstudio-server:1", (uc-cdis/hatchery#1)
  - "env": { (uc-cdis/hatchery#1)
  - "DISABLE_AUTH": "true" (uc-cdis/hatchery#1)
  - }, (uc-cdis/hatchery#1)
  - "args": [], (uc-cdis/hatchery#1)
  - "path-rewrite": "/", (uc-cdis/hatchery#1)
  - "use-tls": "false", (uc-cdis/hatchery#1)
  - "ready-probe": "/" (uc-cdis/hatchery#1)
  - } (uc-cdis/hatchery#1)
  - ``` (uc-cdis/hatchery#1)
- pidgin
  - added required configuration variable "API_HEALTH_URL" for status endpoint (uc-cdis/pidgin#31)
  - use the latest version of the python 3 base image (uc-cdis/pidgin#29)
- portal
  - InCommon login needs fence 4.8.0 (uc-cdis/portal#595)
  - Need Guppy 0.3.7+ to handle large file size (uc-cdis/portal#603)
  - if deployed with "useArboristUI": true, needs Arborist's GET /auth/mapping endpoint to be exposed via revproxy. (useArboristUI false by default) (uc-cdis/portal#592)
  - New optional portal config field called `adminPreAppliedFilters`. Goes in data/config/*.json. (uc-cdis/portal#576)
  - For a "cert" if the required document is a quiz, then hasCorrectAnswers should be true, otherwise false. (uc-cdis/portal#571)
  - Windmill lists unmapped files by checking Indexd. Before centralized auth's deployment, a file was determined to be unmapped by checking if the acl field was null. After centralized auth, we use the authz field instead. The new manifest configuration setting `use_indexd_authz` in the global block specifies which of the two fields should be used. This field defaults to false. (uc-cdis/portal#570)
  - Deployed in https://qa-canine.planx-pla.net with fence version < 3, index authz field set to false (uc-cdis/portal#570)
  - (by default). (uc-cdis/portal#570)
  - Deployed in https://qa-brain.planx-pla.net with fence version > 3, index authz field set to true. (uc-cdis/portal#570)
  - New portal configuration and images for AnVIL (uc-cdis/portal#569)
  - Requires a url to be specified in `dataExplorerConfig` called `terraExportURL` (uc-cdis/portal#568)
  - This should be `bvdp-saturn-${env}.appspot.com/#import-data`, where `${env}` is `dev`, `staging`, or `prod` (uc-cdis/portal#568)
  - Needs a version of cloud-automation that exposes arborist's `auth/proxy` endpoint (uc-cdis/portal#565)
  - fixed DAT requires Guppy >= 0.3.1 (uc-cdis/portal#563)
  - DAT requires Guppy >= 0.3.0 (uc-cdis/portal#561)

File: gen3.theanvil.io/manifest.json [hunk 1]
```diff
@@ -67,7 +67,7 @@
     "sidecar": {
       "cpu-limit": "1.0",
       "memory-limit": "512Mi",
-      "image": "quay.io/cdis/gen3fuse-sidecar:chore_sidecar",
+      "image": "quay.io/cdis/gen3fuse-sidecar:2020.02",
```
Comment (frickjack) [inline gen3.theanvil.io/manifest.json]: please bump the sidecar to `0.1.5` - has a fix for fuse hammering the squid

File: gen3.theanvil.io/manifest.json [hunk 2]
```diff
@@ -7,32 +7,32 @@
     "autodeploy": "yes"
   },
   "versions": {
-    "arborist": "quay.io/cdis/arborist:deploy_anvil-staging",
+    "arborist": "quay.io/cdis/arborist:2020.02",
     "arranger-adminapi": "quay.io/cdis/arranger-server:master",
     "arranger-dashboard": "quay.io/cdis/arranger-dashboard:master",
```
Comment (frickjack) [inline gen3.theanvil.io/manifest.json]: you can remove these `arranger-*`

Comment (frickjack) [inline gen3.theanvil.io/manifest.json]: you can remove these `arranger-*`

File: gen3.theanvil.io/manifest.json [hunk 3]
```diff
@@ -7,32 +7,32 @@
...
+    "fence": "quay.io/cdis/fence:2020.02",
     "fluentd": "fluent/fluentd-kubernetes-daemonset:v1.2-debian-cloudwatch",
-    "guppy": "quay.io/cdis/guppy:0.3.0",
-    "hatchery": "quay.io/cdis/hatchery:0.1.0",
-    "indexd": "quay.io/cdis/indexd:2.0.0",
-    "manifestservice": "quay.io/cdis/manifestservice:master",
-    "peregrine": "quay.io/cdis/peregrine:master",
-    "pidgin": "quay.io/cdis/pidgin:1.1.0",
-    "portal": "quay.io/cdis/data-portal:2.17.0",
+    "guppy": "quay.io/cdis/guppy:2020.02",
```
Comment (frickjack) [inline gen3.theanvil.io/manifest.json]: Please verify the `etlMapping.yaml` and `portal/gitops.json` match what's working in staging - even though `guppy` is in the manifest - it's not actually running in production for some reason

File: gen3.theanvil.io/manifest.json [hunk 4]
```diff
@@ -7,32 +7,32 @@
...
+    "hatchery": "quay.io/cdis/hatchery:2020.02",
+    "indexd": "quay.io/cdis/indexd:2020.02",
+    "manifestservice": "quay.io/cdis/manifestservice:2020.02",
+    "peregrine": "quay.io/cdis/peregrine:2020.02",
+    "pidgin": "quay.io/cdis/pidgin:2020.02",
+    "portal": "quay.io/cdis/data-portal:2020.02",
     "revproxy": "quay.io/cdis/nginx:1.15.5-ctds",
```
Comment (frickjack) [inline gen3.theanvil.io/manifest.json]: We need to deploy `revproxy:1.17...` to include the new security baseline and web application firewall

**Comment by PlanXCyborg** [comment]
# gen3.theanvil.io/manifest.json
## :warning: Services on branch
- wts
## Deployment changes
- fence
  - Fence does NOT handle `c999` the same as before. If using dbgap sync w/ consent code parsing and indexd records have acls with just `phsid` to represent objects someone w/ `c999` should be able to see, you will need to ensure that the records also have the phsid.consent_code as well or users will lose access (uc-cdis/fence#702)
  - Fence has a new configuration to support more dbGaP syncing options: if you require "Common Exchange Area" data access, review the new configuration and supply necessary values (uc-cdis/fence#702)
  - Fence config ENABLED_IDENTITY_PROVIDERS section deprecated and replaced by LOGIN_OPTIONS section to allow enabling several login options for the same provider (backwards compatible) (uc-cdis/fence#703)
  - Requires manual work in Arborist DB to fill in the empty `authz_provider` fields (uc-cdis/fence#696)
  - new cloud automation job for database migration https://github.com/uc-cdis/cloud-automation/pull/977 (uc-cdis/fence#694)
- guppy
  - Commons admins can specify a selectedValues filter using the `adminPreAppliedFilters` field in a portal config file using Guppy syntax. The filter is applied to to the records displayed before any user-applied filters, and can be specified independently in the dataExplorerConfig and fileExplorerConfig blocks. Example: (uc-cdis/guppy#56)
  - ``` (uc-cdis/guppy#56)
  - "adminAppliedPreFilters": { (uc-cdis/guppy#56)
  - "project_id": { (uc-cdis/guppy#56)
  - "selectedValues": ["jnkns-jenkins"] (uc-cdis/guppy#56)
  - } (uc-cdis/guppy#56)
  - } (uc-cdis/guppy#56)
  - ``` (uc-cdis/guppy#56)
- hatchery
  - New `pull_policy` option is available for `hatchery.containers` entries, eg: (uc-cdis/hatchery#1)
  - ``` (uc-cdis/hatchery#1)
  - { (uc-cdis/hatchery#1)
  - "target-port": 8787, (uc-cdis/hatchery#1)
  - "cpu-limit": "1.0", (uc-cdis/hatchery#1)
  - "memory-limit": "512Mi", (uc-cdis/hatchery#1)
  - "pull_policy": "Always", (uc-cdis/hatchery#1)
  - "name": "R Studio", (uc-cdis/hatchery#1)
  - "image": "heliumdatastage/rstudio-server:1", (uc-cdis/hatchery#1)
  - "env": { (uc-cdis/hatchery#1)
  - "DISABLE_AUTH": "true" (uc-cdis/hatchery#1)
  - }, (uc-cdis/hatchery#1)
  - "args": [], (uc-cdis/hatchery#1)
  - "path-rewrite": "/", (uc-cdis/hatchery#1)
  - "use-tls": "false", (uc-cdis/hatchery#1)
  - "ready-probe": "/" (uc-cdis/hatchery#1)
  - } (uc-cdis/hatchery#1)
  - ``` (uc-cdis/hatchery#1)
- pidgin
  - added required configuration variable "API_HEALTH_URL" for status endpoint (uc-cdis/pidgin#31)
  - use the latest version of the python 3 base image (uc-cdis/pidgin#29)
- portal
  - InCommon login needs fence 4.8.0 (uc-cdis/portal#595)
  - Need Guppy 0.3.7+ to handle large file size (uc-cdis/portal#603)
  - if deployed with "useArboristUI": true, needs Arborist's GET /auth/mapping endpoint to be exposed via revproxy. (useArboristUI false by default) (uc-cdis/portal#592)
  - New optional portal config field called `adminPreAppliedFilters`. Goes in data/config/*.json. (uc-cdis/portal#576)
  - For a "cert" if the required document is a quiz, then hasCorrectAnswers should be true, otherwise false. (uc-cdis/portal#571)
  - Windmill lists unmapped files by checking Indexd. Before centralized auth's deployment, a file was determined to be unmapped by checking if the acl field was null. After centralized auth, we use the authz field instead. The new manifest configuration setting `use_indexd_authz` in the global block specifies which of the two fields should be used. This field defaults to false. (uc-cdis/portal#570)
  - Deployed in https://qa-canine.planx-pla.net with fence version < 3, index authz field set to false (uc-cdis/portal#570)
  - (by default). (uc-cdis/portal#570)
  - Deployed in https://qa-brain.planx-pla.net with fence version > 3, index authz field set to true. (uc-cdis/portal#570)
  - New portal configuration and images for AnVIL (uc-cdis/portal#569)
  - Requires a url to be specified in `dataExplorerConfig` called `terraExportURL` (uc-cdis/portal#568)
  - This should be `bvdp-saturn-${env}.appspot.com/#import-data`, where `${env}` is `dev`, `staging`, or `prod` (uc-cdis/portal#568)
  - Needs a version of cloud-automation that exposes arborist's `auth/proxy` endpoint (uc-cdis/portal#565)
  - fixed DAT requires Guppy >= 0.3.1 (uc-cdis/portal#563)
  - DAT requires Guppy >= 0.3.0 (uc-cdis/portal#561)

**Pull request merged**